internal/core/future.rs

```diff
@@ -150,29 +150,89 @@ unsafe impl<T: Send> Send for JoinHandle<T> {}
+/// the Slint runtime can't drive the Tokio runtime.
+/// * Tokio futures require regular yielding to the Tokio runtime for fairness, a constraint that also can't be met by Slint.
+/// * Tokio's [current-thread schedule](https://docs.rs/tokio/latest/tokio/runtime/index.html#current-thread-scheduler)
+/// cannot be used in Slint main thread, because Slint cannot yield to it.
+///
+/// To addresse these constraints, use [async_compat](https://docs.rs/async-compat/latest/async_compat/index.html)'s [Compat::new()](https://docs.rs/async-compat/latest/async_compat/struct.Compat.html#method.new)
+/// to implicitly allocate a shared, multi-threaded Tokio runtime that will be used for Tokio futures.
+///
+/// The following little example demonstrates the use of Tokio's [`TcpStream`](https://docs.rs/tokio/latest/tokio/net/struct.TcpStream.html) to
+/// read from a network socket. The entire future passed to `spawn_local()` is wrapped in `Compat::new()` to make it run:
+///
+/// ```rust,no_runt
+/// // A dummy TCP server that once reports "Hello World"
 /// # i_slint_backend_testing::init_integration_test_with_mock_time();
-/// // In your main function, create a runtime that runs on the other threads
-/// let tokio_runtime = tokio::runtime::Runtime::new().unwrap();
+/// use std::io::Write;
 ///
-/// // ...
-/// // Within the UI thread (for example in a callback handler)
-/// slint::spawn_local(async move {
-///     let result = tokio_runtime.spawn(async move {
-///         // This code is running on the Tokio runtime's thread, you can await futures that depends on Tokio here.
-///         42
-///     }).await.unwrap();
-///     // now we are back on the UI thread so we can do something with the result on the UI thread
-///     assert_eq!(result, 42);
-///     # slint::quit_event_loop();
-/// }).unwrap();
-/// # slint::run_event_loop().unwrap();
+/// let listener = std::net::TcpListener::bind("127.0.0.1:0").unwrap();
+/// let local_addr = listener.local_addr().unwrap();
+/// let server = std::thread::spawn(move || {
+///     let mut stream = listener.incoming().next().unwrap().unwrap();
+///     stream.write("Hello World".as_bytes()).unwrap();
+/// });
+///
+/// let slint_future = async move {
+///     use tokio::io::AsyncReadExt;
+///     let mut stream = tokio::net::TcpStream::connect(local_addr).await.unwrap();
+///     let mut data = Vec::new();
+///     stream.read_to_end(&mut data).await.unwrap();        
+///     assert_eq!(data, "Hello World".as_bytes());
+///     slint::quit_event_loop().unwrap();
+/// };
+///
+/// // Wrap the future that includes Tokio futures in async_compat's `Compat` to ensure
+/// // presence of a Tokio run-time.
+/// slint::spawn_local(async_compat::Compat::new(slint_future)).unwrap();
+///
+/// slint::run_event_loop_until_quit().unwrap();
+///
+/// server.join().unwrap();
+/// ```
+///  
+/// Alternatively, if your application uses `#[tokio::main]` to create a Tokio run-time, wrap the call to enter the Slint
```

Comment:
I still don't think we should recommend that.
There should be a big **it is not recommended** or something like that.

Reply:
You're right. I'll rephrase with emphasis.

Reply:
What do you think about this new version?

Overall, I'm a bit torn. I see that entering a runtime that's just there blocked seems wasteful. On the other hand, with tokio::main the defaults are decided by tokio, with async-compat the allocation and defaults are with another separate crate that needs to be added as a dependency. The runtime result is - I suppose - the same: any actual work is - in both cases - delegated to worker threads of a global runtime (either allocated by tokio in tokio::main or by async-compat in the global variable).

Reply:
I think it is better like this.

I would probably have left the original example that creates a toklio runtime manually and say that it can be simplified with the use of the async-compat crate.

Reply:
Good idea. I started out wanting to have these tests as doctests, but I realise that's not possible due to our platform singleton right now. But now that the tests are separate, I think you're right: I'll go back to the original example, but use the simplified async_compat way right away. Then it's less noisy, and I'll leave out the example with `block_in_place` and merely mention it.

Reply:
Ok, I looked back at the example and I prefer the new one where tokio futures are _actually_ used and where the application of `Compat::new()` makes sense, as opposed to the `rt.spawn(async move { 42 })` example ;)